### PR TITLE
Fixes for nexus-push

### PR DESCRIPTION
**Pull request by lubomir**
This is probably best reviewed commit by commit.

There are a couple fixes in the code to avoid shadowing existing variables and using correct arguments.

The ZIP splitter is modified to not try adding empty directories to created archives. This was causing issues with having empty filename. The file count and accumulated file size is reset after each part is written.

The temporary directory with parts is deleted once the script finishes.

When uploading, a line is printed for each part to give some idea of progress.

The code to remove `maven-repository` from path is modified to skip it even multiple leading path components that contain that substring.

Now for the big changes:

Instead of manual modification of groups, the promote is changed to use API call. The repo status is checked after the call to report any problems. Issues detected in promotion and closing are reported to user.

Similarly rollback works by dropping the repo via API. User still has to find repo ID from web UI.

Since the code for group manipulation is not used anymore, the entire file is dropped.

### CODE REVIEW

File: rcm_nexus/command.py [hunk 1]
```diff
@@ -2,19 +2,22 @@
...
-import re
 import click
+import shutil
 import tempfile
 
 RELEASE_GROUP_NAME = 'product-ga'
 TECHPREVIEW_GROUP_NAME = 'product-techpreview'
 PRERELEASE_GROUP_NAME = 'product-earlyaccess'
 
+EARLY_ACCESS_PROFILE_ID = "537461ce29c60"
```
Comment (jdcasey) [inline rcm_nexus/command.py]: Would these be better suited to the configuration file?

Comment (lubomir) [reply rcm_nexus/command.py]: Probably yes. As far as I understand it, the values are not expected to change and are product independent. But you're right that config file is probably a better place for it.

**Review by jdcasey — APPROVED** [review]
Looks good to me. I had just one small quibble with the location of staging profile ids, but it'll work either way and while it's technically internal data I don't think it's actionable in any security sense.

**Comment by lubomir** [comment]
Last two commits move the IDs to config file and include LICENSE in source tarball (so that it can be installed in RPM package).

**Comment by jdcasey** [comment]
LGTM

**Pull request merged**